### PR TITLE
[IMP] Purchase search product

### PR DESCRIPTION
**Pull request by nicolasj77**
Dans un bon d'achat il n'est plus possible de commander un article qui n'est pas coché "Peux être acheté"

### CODE REVIEW

File: purchase_custom_search_product/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+from . import models
```
Comment (robinkeunen) [inline purchase_custom_search_product/__init__.py]: missing eol and unnecessary encoding line.

File: purchase_custom_search_product/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "license": "AGPL-3",
+    "website": "www.coopiteasy.be",
+    "description": """
+        Do not allow to add product mark as 'Cannot be buy'
+    """,
+    "data": [
+        'views/purchase_view.xml',
+    ],
+    'installable': True,
+}
```
Comment (robinkeunen) [inline purchase_custom_search_product/__openerp__.py]: Missing eol

File: purchase_custom_search_product/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# -*- coding: utf-8 -*-
```
Comment (robinkeunen) [inline purchase_custom_search_product/models/__init__.py]: No necessary here.

File: purchase_custom_search_product/views/purchase_view.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+            <field name="inherit_id" ref="purchase.purchase_order_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//notebook//page/field[@name='order_line']//field[@name='product_id']"
+                       position="attributes">
+                    <attribute name="domain">[('purchase_ok', '=', True)]</attribute>
+                </xpath>
+            </field>
+        </record>
+    </data>
+</odoo>
```
Comment (robinkeunen) [inline purchase_custom_search_product/views/purchase_view.xml]: missing eol

File: purchase_custom_search_product/views/purchase_view.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<odoo>
+    <data>
+        <record id="purchase_order_form_inherited" model="ir.ui.view">
+            <field name="name">purchase.order.form.inherit</field>
+            <field name="model">purchase.order</field>
+            <field name="inherit_id" ref="purchase.purchase_order_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//notebook//page/field[@name='order_line']//field[@name='product_id']"
+                       position="attributes">
+                    <attribute name="domain">[('purchase_ok', '=', True)]</attribute>
```
Comment (robinkeunen) [inline purchase_custom_search_product/views/purchase_view.xml:11]: @remytms is there a way to add elements to a domain instead of overriding it?

**Review by robinkeunen — APPROVED** [review]
Looks good to me, I just asked a question to Rémy to make sure.

File: purchase_custom_search_product/__init__.py (hunk 1 above)
Comment (nicolasj77) [reply purchase_custom_search_product/__init__.py]: ok

File: purchase_custom_search_product/__openerp__.py (hunk 2 above)
Comment (nicolasj77) [reply purchase_custom_search_product/__openerp__.py]: ok

File: purchase_custom_search_product/views/purchase_view.xml (hunk 4 above)
Comment (nicolasj77) [reply purchase_custom_search_product/views/purchase_view.xml]: ok

File: purchase_custom_search_product/views/purchase_view.xml (hunk 5 above)
Comment (remytms) [reply purchase_custom_search_product/views/purchase_view.xml:11]: @robinkeunen perhaps this example for class : 
```xml
<attribute name="class" add="col-sm-5 o_grid" separator=" "/>
```
But for a domain it's not so easy. So I think there is no better way than replacing it. :/

Sources : https://www.odoo.com/zh_CN/forum/help-1/question/xpath-add-attribute-value-instead-of-replacing-it-125230

File: purchase_custom_search_product/models/__init__.py (hunk 3 above)
Comment (remytms) [reply purchase_custom_search_product/models/__init__.py]: Yes better to have an empty file if there is no python code in it.

Comment (nicolasj77) [reply purchase_custom_search_product/models/__init__.py]: Done

File: purchase_custom_search_product/views/purchase_view.xml (hunk 5 above)
Comment (nicolasj77) [reply purchase_custom_search_product/views/purchase_view.xml:11]: So, something to change ?

Comment (robinkeunen) [reply purchase_custom_search_product/views/purchase_view.xml:11]: No, it's ok. I checked `purchase.order.line` and there are no additional domain. In fact, it is already 
```
    product_id = fields.Many2one(... domain=[('purchase_ok', '=', True)] ... )
```
so I don't understand why you'd need to overload it.

**Review by robinkeunen — APPROVED** [review]

File: purchase_custom_search_product/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "version": "1.0",
+    "depends": [
+        'purchase',
+    ],
+    "author": "Coop IT Easy SCRLfs",
+    "license": "AGPL-3",
+    "website": "www.coopiteasy.be",
+    "description": """
+        Do not allow to add product mark as 'Cannot be buy'
+    """,
```
Comment (houssine78) [inline purchase_custom_search_product/__openerp__.py:16]: typo => marked

**Review by houssine78 — APPROVED** [review]

**Pull request merged**